### ISSUE TITLE
Marginnotes don't appear in margin when in a list

### ISSUE THREAD

**Issue by dmerrick** [edwardtufte/tufte-css#117]
I have a marginnote in an unordered list, it looks like this:

```html
<ul>
  <li>Tufte-inspired stylesheet by <a href="https://edwardtufte.github.io/tufte-css/">tufte-css</a><label for="mn-6" class="margin-toggle">⊕</label><input type="checkbox" id="mn-6" class="margin-toggle"><span class="marginnote">Seriously, check out <a href="https://en.wikipedia.org/wiki/Edward_Tufte">Edward Tufte</a> and his work on information design</span></li>
</ul>
```

Unfortunately, the note text does not appear in the margin. Oddly, when the viewport is shrunk, I can click the "expand" icon and see the note inline, but when I restore the viewport to the original size, the marginnote disappears.

This is the same for sidenotes and unordered lists as well.

**Comment by crmackay** [comment]
I put your code above in between two paragraphs, and the marginnote showed up...its a bit misaligned, which should be fixed, but I can see it. Can you provide a bit more of a minimal example and perhaps a screen shot of the problem your seeing?

![image](https://user-images.githubusercontent.com/3801862/30882454-ffb994fe-a2d6-11e7-8a86-bd6b84520a3e.png)

**Comment by dmerrick** [comment]
Thanks for responding, @crmackay!

Here's an example of this happening: https://www.dana.lol/2017/09/26/thanks/

**Comment by dmerrick** [comment]
Interestingly, if I wrap the `<ul>` in a `<p>`, I get this:

![demo](https://i.imgur.com/SMz9krb.png)

The `*`s come from the fact that this is [written in Markdown](https://github.com/dmerrick/danalol/blob/master/source/2017-09-26-thanks.html.md.erb).

**Comment by crmackay** [comment]
so the reason appears to be because you do not have your `ul` and `ol` elements inside a `<section>` ... `</section>` element.

**Comment by dmerrick** [comment]
Thanks, I was wondering if that was the case. I'm not a CSS guy so it's all kind of a mystery to me!

I added `<section>` tags and the sidenotes now appear, yay!

The last issue seems to be that the notes aren't aligned with the other notes:

![example](https://i.imgur.com/GMYQ5aa.png)

**Comment by jez** [comment]
I think lists in general need a bit of TLC:

- Without wrapping list item bodies in `<p>` tags, the line height doesn't get set properly.
- With wrapping list item bodies in `<p>` tags, you get too much vertical spacing between list items, and the width of lists is constrained to 55% of the section width.
- Either way (`<p>` or no `<p>`), the margin notes don't show up at the correct alignment.

**Comment by daveliepmann** [comment]
I agree that lists need some TLC. I will not have time to look at anything Tufte CSS-related for a couple months at a minimum. PRs welcome, as long as they involve extensive testing to make sure we don't introduce any regressions. :)

**Comment by jez** [comment]
@dmerrick I have a fix for this in #118. If you could try it out and report whether it fixes the layouting issues you were getting, that'd be great!

**Comment by dmerrick** [comment]
Seems to work for me!
